### DISCUSSION TITLE
Apply a Loading Rate (Force) as a Boundary condition

### DISCUSSION THREAD

**Discussion by Vandenbg** [idaholab/moose#24413] [Q&A General]
I am currently working on a spherical indentation model in MARMOT and until now I have been using displacement control. I would like to compare my results to different applied loads over time,  i.e. 20mn/s or 80mn/s. This should be set up the same as MOOSE. So far I have tried two different methods, Pressure BC and Neumann BC. I have been unsuccessful on both accounts in getting my indenter tip to budge at all. 

This is what I have been using. 
```
  [push_down]
    type = ParsedFunction
    expression = 'if(t<2, -(t/0.005)*3.2e-6, -0.001280)'
  []
  ...
  [topy]
    type = FunctionDirichletBC
    variable = disp_y
    boundary = '202'
    function = push_down
  []
```

This is what I tried for Neumann BC. I attempted several functions, each in the same form but changing the 0.020 to higher numbers. 
```
[topy]
  type = FunctionNeumannBC
  variable = disp_y
  boundary = '202'
  function = t*0.020
[]
```

Here is what I attempted with Pressure BC. Again, I changed the 0.020 value to a range of values with not avail.
```
[Pressure]
  [topy]
    boundary = '202'
    function = t*0.020
    displacements = 'disp_y'
  []
[]
```

Please let me know If I am doing this correctly, or how I should go about applying a specific loading rate to a boundary. Thank you!

**Comment by GiudGiud** [comment]
So you are applying both a dirichlet and a load on the same boundary?

**Reply by Vandenbg** [reply]
Yes, for boundary 202, disp_x and disp_z I am applying dirichlet conditions of 0, and for disp_y I am using what I described above.

**Reply by GiudGiud** [reply]
For disp_y, you have a dirichlet above as well

**Reply by Vandenbg** [reply]
That is correct, it is what I have been using up until this point. The three examples above were all used independently. All three were not used at the same time.

**Reply by GiudGiud** [reply]
how many time steps did you take? what was the value of 't' in the end

**Reply by Vandenbg** [reply]
In both cases, I let the simulation run with no displacement for roughly 0.5 seconds.

**Comment by sapitts** [comment]
@recuero and @jiangwen84 do either of you have any insights on modeling nanoindentation with a prescribed load (rather than displacement control)?

**Reply by recuero** [reply]
The methodology I'd use would be something like: 

0. Make sure to remove any possible rigid body motion in the indenter
1. Use displacement control to move the indenter down (Dirichlet BC)
2. Establish contact to some extent. 
3. Switch to load control, i.e. remove the Dirichlet BC and use controls to impose a load/pressure gradually.
4. Make sure the transition between displacement control and load control is smooth. You do that by computing a reaction force for the displacement control part and then applying a consistent initial pressure value.

**Reply by Vandenbg** [reply]
Thank you for this suggestion. I will attempt this method. Have I set up the Pressure boundary condition correctly, would the reaction force value be placed where 0.02 is in the function?

```
[Pressure]
  [topy]
    boundary = '202'
    function = t*0.020 #<-- here?
    displacements = 'disp_y'
  []
[]
```

**Reply by recuero** [reply]
Yes, but note that you'll have to sum all the reaction forces and then come up with an initial pressure value that'll have a similar overall loading effect. It doesn't have to be exact, but it has to: 1) Keep precluding rigid body motion and 2) Not deviate from equilibrium too much, i.e. the system needs to be able to converge.
